packages/stark-ui/src/modules/table/components/table.component.spec.ts

```diff
@@ -236,6 +239,21 @@ describe("TableComponent", () => {
 			expect(component.displayedColumns.unshift).not.toHaveBeenCalled();
 		});
 
+		it("should assign right value to _showRowIndex when showRowIndex changes and adapt displayedColumns", () => {
+			spyOn(component.displayedColumns, "unshift");
+			hostComponent.showRowIndex = true;
+			hostFixture.detectChanges();
+			expect(component._showRowIndex).toBe(true);
+			expect(component.displayedColumns.unshift).toHaveBeenCalledTimes(1);
+			expect(component.displayedColumns.unshift).toHaveBeenCalledWith("rowIndex");
```

Comment:
I think the test should focus on the functionality rather than on the implementation. In this case we should not test that `unshift` was called because this is an implementation detail. The test should focus on the removal and the changes on the `displayedColumns` 😉

Reply:
Actually, I applied the same logic than the one we applied for `multiSelect` when we check if the checkbox is displayed or not by checking the `displayedColumns`. 
I'll try to improve this 😊

Reply:
Yep, that's fine... I'm just saying that we should avoid testing the internal functions we call such as `unshift` cause maybe we might change that later on and our test will fail even though the functionality is still correct :s